### PR TITLE
Enable proofs on CI for proof enabled builds

### PR DESCRIPTION
**Pull request by ajreynol**

### CODE REVIEW

File: src/theory/builtin/proof_checker.cpp [hunk 1]
```diff
@@ -422,7 +422,8 @@ Node BuiltinProofRuleChecker::checkInternal(PfRule id,
            || id == PfRule::THEORY_PREPROCESS_LEMMA
            || id == PfRule::THEORY_EXPAND_DEF || id == PfRule::WITNESS_AXIOM
            || id == PfRule::THEORY_LEMMA || id == PfRule::THEORY_REWRITE
-           || id == PfRule::TRUST_SUBS || id == PfRule::TRUST_SUBS_MAP)
+           || id == PfRule::TRUST_REWRITE || id == PfRule::TRUST_SUBS
+           || id == PfRule::TRUST_SUBS_MAP)
```
Comment (nafur) [inline src/theory/builtin/proof_checker.cpp:426]: Did you mean to include this change? It seems unrelated to the PR.

**Review by nafur — APPROVED** [review]
LGTM, except for one comment.

File: src/theory/builtin/proof_checker.cpp (hunk 1 above)
Comment (ajreynol) [reply src/theory/builtin/proof_checker.cpp:426]: Yes, it's actually required to get CI to pass. Making this PR revealed that master had not taken this change from proof-new yet.

**Comment by ajreynol** [comment]
FYI, we need to fix a debug failure on regress1 before merging this. @HanielB is investigating.  Also, we should wait before the nightlies pass again, which @mpreiner is working on.

File: test/regress/run_regression.py [hunk 2]
```diff
@@ -382,6 +382,15 @@ def run_regression(unsat_cores, proofs, dump, use_skip_return_code,
         if extra_command_line_args:
             command_line_args_configs.append(all_args +
                                              extra_command_line_args)
+            
+        extra_command_line_args_proof = []
+        if proofs and re.search(r'^(unsat|valid)$', expected_output):
+            if '--no-check-proofs' not in all_args and \
+               '--check-proofs' not in all_args:
```
Comment (4tXJ7f) [inline test/regress/run_regression.py:389]: IIRC, we were combining proofs and unsat cores in the same call. This change will call the solver three times instead of two (at least on the unsat benchmarks). Are we ok with this?

File: test/regress/run_regression.py [hunk 3]
```diff
@@ -473,7 +482,7 @@ def main():
 
     timeout = float(os.getenv('TEST_TIMEOUT', 600.0))
 
-    return run_regression(args.enable_proof, args.with_lfsc, args.dump,
+    return run_regression(args.enable_proof, args.enable_proof, args.dump,
```
Comment (4tXJ7f) [inline test/regress/run_regression.py:485]: I think that it would be good to remove the separate argument for unsat cores from `run_regression()` and the `--with-lfsc` from the argument parser above.

**Review by 4tXJ7f — CHANGES_REQUESTED** [review]

File: test/regress/run_regression.py (hunk 2 above)
Comment (ajreynol) [reply test/regress/run_regression.py:389]: Yes, this was the intention.  In the future, unsat cores and proofs will be fairly disjoint (apart from preprocessing).

**Comment by ajreynol** [comment]
Note this is subsumed by https://github.com/CVC4/CVC4/pull/6088.
